u4-database/features/uni-controller.js

```diff
@@ -0,0 +1,54 @@
+const { db } = require('../_services/firebase-initialized')
+
+const iterationLists = async (req, res) => {
+  const query = db.collection('iterations').get()
+  const iterations = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-iteration-list', { iterations })
+}
+
+const iterationDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('iterations').doc(code).get()
+  const iteration = (await query).data()
+
+  res.render('uni-iteration-details', { iteration })
+}
+
+const studentDetails = async (req, res) => {
+  const code = req.params.code
+  const query = db.collection('students').doc(code).get()
+  const student = (await query).data()
+
+  res.render('uni-student-details', { student })
+}
+
+const courseLists = async (req, res) => {
+  const query = db.collection('iterations')
+    .where('year', '==', 2021)
+    .where('semester', '==', 1)
+    .get()
+  const courses = (await query).docs.map(doc => doc.data())
+
+  res.render('uni-course-list', { courses })
+}
+
+const courseDetails = async (req, res) => {
+  const code = req.params.code
```

Comment:
เพื่อหลีกเลี่ยงความสับสน อาจเป็นการดีกว่าที่จะตั้งชื่อ `courseCode` นี้ นี่แสดงถึงรหัสเช่น "231370" อินสแตนซ์ก่อนหน้าของ `code` แสดงถึงโค้ดการวนซ้ำ เช่น "231370.20s1"

To avoid confusion, it might be better to name this `courseCode`. This represents a code like "231370". Previous instances of `code` represented an iteration code like "231370.20s1".